dev-lang/luajit/luajit-2.1.0_beta3_p20220613.ebuild

```diff
@@ -31,6 +31,21 @@ IUSE="lua52compat static-libs"
 
 S="${WORKDIR}/LuaJIT-${GIT_COMMIT}"
 
+src_configure() {
+	tc-export_build_env
+
+	# You need to use a 32-bit toolchain to build for a 32-bit architecture.
+	# Some 64-bit toolchains (like amd64 and ppc64) usually have multilib
+	# enabled, allowing you to build in 32-bit with -m32. This won't work in all
+	# cases, but it will otherwise just break, so it's worth trying anyway. If
+	# you're trying to build for 64-bit from 32-bit, then you're screwed, sorry.
+	if tc-is-cross-compiler && [[ $($(tc-getBUILD_CPP) -P - <<< __INTPTR_WIDTH__) = 64 &&
```

Comment:
Despite vapier's protests, I think we should really implement https://bugs.gentoo.org/328401 at some point.

I'm rather curious about why this limitation exists to begin with, but of course, I believe you, and given how weird Luajit is, I'm not surprised at all.

EDIT: Ah, https://github.com/LuaJIT/LuaJIT/issues/664. Please add a link to that in the comment.

Reply:
I did consider adding an eclass function until I saw how little code it required. I'm still happy to do that. I think our ecosystem is mature enough that the concerns vapier had are not so relevant now. He's hardly around to call the shots anyway. What I've written is also less hacky than the initial suggestions. Will update shortly.

Reply:
Done that. I went with `__SIZEOF_POINTER__` in the end, which is in bytes rather than bits. If you're happy, I'll send this to the list.